frontend/package.json

```diff
@@ -20,13 +20,15 @@
     "@angular/common": "^5.0.0",
     "@angular/compiler": "^5.0.0",
     "@angular/core": "^5.0.0",
+    "@angular-devkit/core": "^0.0.29",
```

Comment:
why were these 2 dependencies added?

Reply:
It raised some errors without these dependencies:
```
module.js:478
    throw err;
    ^

Error: Cannot find module '@angular-devkit/core'
    at Function.Module._resolveFilename (module.js:476:15)
    at Function.Module._load (module.js:424:25)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/src/app/node_modules/@angular-devkit/schematics/src/tree/virtual.js:10:16)
    at Module._compile (module.js:577:32)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
    at Function.Module._load (module.js:445:3)
```

Reply:
Which is why you use yarn which ensures exact dependencies as the one which previously compiled this project are used :P

Reply:
@yashsriv Okay, I reverted back to using yarn :p

Reply:
@yashsriv Doesn't the `package-lock.json` file created by npm do the same thing as `yarn.lock` now? Is it required anymore really? @yashsriv

Reply:
@abdatta Technically not. But this is like a 3-4 year old project still using angular 5. If someone brings things up-to-date, they can get rid of yarn/yarn.lock and shift to using npm+package-lock.json.

Reply:
Makes sense. I created an issue to upgrade the angular and other packages of the frontend. Once that's done, we can hopefully reduce one devDependency :P